apps/acls/serializers/base.py

```diff
@@ -70,6 +70,13 @@ def set_action_choices(self):
             return
         if not settings.XPACK_LICENSE_IS_VALID:
             field_action._choices.pop(ActionChoices.review, None)
+        if not (
+            settings.XPACK_LICENSE_IS_VALID and
+            settings.XPACK_LICENSE_EDITION_ULTIMATE and
+            settings.FACE_RECOGNITION_ENABLED
+        ):
+            field_action._choices.pop(ActionChoices.face_verify, None)
+            field_action._choices.pop(ActionChoices.face_online, None)
         for choice in self.Meta.action_choices_exclude:
             field_action._choices.pop(choice, None)
 
```

Comment:
There isn't much in terms of irregularities, potential issues, or optimization to suggest here. The provided code looks relatively standard with minor improvements suggested in comments about license conditions.

The changes you've noticed can be summarized as:

1) If `settings.XPACK_LICENSE_IS_VALID` is false then remove an action named review from list choices

2) Remove face verification / online options if `XPACK_LICENSE_EDITION_ULTIMATE`, `FACE_RECOGNITION_ENABLED` fields are not valid and enabled

However, these updates align well with general best practices related to licenses and authentication checks at this time. It's always good practice that all users should have proper access and usage permissions based on their roles or identities. 

In summary, it appears the function will run without any errors; though some adjustments have been recommended considering specific environment configurations (like XPACK license). This could be used for further validation before production deployment or for additional configuration checking post-production release testing phases. Please let me know if there needs more explanation or clarification needed!